### PR TITLE
Add required action for phone validation

### LINKED ISSUE #16: Introduce verification step
A phone number should be verified before the user should be able to use it as second factor. This is similar to TOTP, where the user needs to enter the first OTP, before TOTP becomes active as a second factor.

Procedure:
- User enters a phone number
- SMS with verification code is sent to number
- User enters code
- SMS 2FA is activated

### PR DESCRIPTION
**Pull request by melegiul**
- [x] Add required action for phone validation
- [x] Implement SMS code validation for this action
- [x] Add option to force users to setup 2FA with SMS auth https://github.com/netzbegruenung/keycloak-2fa-sms-authenticator/pull/20/commits/04d640b2d8b8a033055bf5603aad7d604f4b14fd
- [ ] ~Cleaner solution for loading auth configs~ moved to #24 

After phone number submission, the phone is stored in AuthenticationSessionModel and a PhoneValidationRequiredAction is triggered. Phone validation sends SMS code and finally stores users phone number after valid SMS code submission.

TODOs
~When the SMS code validation is aborted without valid code submission, the PhoneNumberRequiredAction is considered as "successful" despite of SMS Authentication could not be setup due to the missing SMS code.
In this case PhoneNumberRequiredAction should be considered as failed.~

Fixes: #16
Fixes: #22
Fixes: #23

### CODE REVIEW

**Comment by svenseeberg** [comment]
Potential solution: at the beginning of step 2 create another required action for step 1. If step 2 is successful, delete the required action for step 1.

File: src/main/java/netzbegruenung/keycloak/authenticator/PhoneValidationRequiredAction.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	@Override
+	public void evaluateTriggers(RequiredActionContext context) {
+	}
+
+	@Override
+	public void requiredActionChallenge(RequiredActionContext context) {
+		context.getAuthenticationSession().addRequiredAction(PhoneNumberRequiredAction.PROVIDER_ID);
+		try {
+			UserModel user = context.getUser();
+			AuthenticatorConfigModel config = context.getRealm().getAuthenticatorConfigByAlias("sms-2fa");
```
Comment (svenseeberg) [inline src/main/java/netzbegruenung/keycloak/authenticator/PhoneValidationRequiredAction.java:59]: We need a more general approach here, as we cannot guarantee that this is always this ID.

Comment (svenseeberg) [reply src/main/java/netzbegruenung/keycloak/authenticator/PhoneValidationRequiredAction.java]: I was thinking that we could provide the config alias with auth notes. However, the processAction method of SmsAuthenticatorSetMobileNumberAction does not know the corresponding config as well. So we're back to square 1.

Comment (svenseeberg) [reply src/main/java/netzbegruenung/keycloak/authenticator/PhoneValidationRequiredAction.java]: We can use it for now for us. But we should probably change this. To merge this PR, we could state the caveat in the README that there needs to be one Execution with the name "sms-2fa" and working SMS API.

Comment (svenseeberg) [reply src/main/java/netzbegruenung/keycloak/authenticator/PhoneValidationRequiredAction.java]: Another dirty hack, but probably less nasty than other solutions, would be getting a config alias from the SQL database for the current realm.

Comment (melegiul) [reply src/main/java/netzbegruenung/keycloak/authenticator/PhoneValidationRequiredAction.java]: > Another dirty hack, but probably less nasty than other solutions, would be getting a config alias from the SQL database for the current realm. Not sure though if the database can be accessed in the scope of an auth provider.

Sounds good, I will try to fetch the config from DB.

**Comment by svenseeberg** [comment]
@melegiul I'd suggest to merge as is and create a new issue for the static Config Alias. I already updated the README in the main branch accordingly.

**Pull request merged**